### PR TITLE
Updated the api link + register/login path + enabled cors

### PR DESCRIPTION
**Pull request by tnjensen**
Fixed the login script, registered users should be able to login.

### CODE REVIEW

**Review by DarioDeLaFuente — COMMENTED** [review]
Why have you added the package-lock.json file?
7,079 changes: 7,064 additions & 15 deletions7,079  
[package-lock.json](https://github.com/NoroffFEU/agency.noroff.dev/pull/396/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519)

**Comment by DarioDeLaFuente** [comment]
@ritugunjan can u approve this pull request.

**Comment by tnjensen** [comment]
@DarioDeLaFuente could you re-review this ? About the package-lock.json, usually I don't exclude that, but in this case there seems to be alot of changes, I'm not sure why. Maybe you can remove it, and it will be recreated when building into production, I think ?

File: src/js/api/auth/login.js [hunk 1]
```diff
@@ -30,7 +28,6 @@ import { dummyApiUrl } from '../constants.js';
    }
  */
 const action = 'users/login';
```
Comment (Fermain) [inline src/js/api/auth/login.js:30]:
```suggestion
const action = 'users/login';
const method = 'POST';
```

File: src/js/api/auth/register.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import { apiUrl } from '../constants.js';
+import { apiPath } from '../constants.js';
 
-const action = 'users';
-const method = 'POST';
+const action = '/users';
+const method = 'post';
```
Comment (Fermain) [inline src/js/api/auth/register.js:4]:
```suggestion
const method = 'POST';
```

File: src/js/api/auth/register.js [hunk 3]
```diff
@@ -12,38 +12,26 @@ const method = 'POST';
  * @returns {Promise<Object>} A Promise that resolves with the registration result if successful.
  * @throws {Error} Throws error if the registration request fails or returns error status.
  */
-
 export async function register(profile) {
```
Comment (Fermain) [inline src/js/api/auth/register.js:15]:
```suggestion
let data, error;
export async function register(profile) {
```

File: src/js/api/auth/register.js [hunk 4]
```diff
@@ -12,38 +12,27 @@ const method = 'POST';
  * @returns {Promise<Object>} A Promise that resolves with the registration result if successful.
  * @throws {Error} Throws error if the registration request fails or returns error status.
  */
-
+let data, error;
```
Comment (Fermain) [inline src/js/api/auth/register.js:15]:
```suggestion
```

File: src/js/api/auth/register.js [hunk 5]
```diff
@@ -12,38 +12,27 @@ const method = 'POST';
...
-    /* switch (response.status) {
-      case 201:
-        const result = await response.json();
-        window.location.replace('/pages/auth/login/index.html');
-        return result;
-      default:
-        throw new Error(`${response.status} ${response.statusText}`);
-    } */
+  const registerURL = apiPath + action;
+
```
Comment (Fermain) [inline src/js/api/auth/register.js:18]:
```suggestion
let data, error;
```

File: src/js/api/auth/register.js [hunk 6]
```diff
@@ -12,38 +12,27 @@ const method = 'POST';
  * @returns {Promise<Object>} A Promise that resolves with the registration result if successful.
  * @throws {Error} Throws error if the registration request fails or returns error status.
  */
-
+let data, error;
 export async function register(profile) {
```
Comment (Fermain) [inline src/js/api/auth/register.js:16]:
```suggestion
export async function register(profile) {
  let data, error;
```

**Review by Fermain — APPROVED** [review]
The code in this PR is not perfect, but in terms of the scope of the task at hand this is good to go.

**Pull request merged**